### PR TITLE
Add interactive graph viewer

### PR DESCRIPTION
**Pull request by mawildoer**
Screenshot 2024-08-22 at 15 35 04

### CODE REVIEW

**Review by iopapamanoglou — APPROVED** [review]
Very nice addition thanks!

File: src/faebryk/exporters/visualize/interactive_graph.py [hunk 1]
```diff
@@ -0,0 +1,214 @@
...
+        except ValueError:
+            return None
+
+        type_name = type(link).__name__
+        link_types.add(type_name)
+
+        return {"data": {"source": source, "target": target, "type": type_name}}
+
+    all_gifs = list(
+        gif for node in get_all_nodes_graph(G) for gif in node.GIFs.get_all()
```
Comment (iopapamanoglou) [inline src/faebryk/exporters/visualize/interactive_graph.py:68]: you can get the gifs with node.GIFs.self.edges instead of get_all to make it graph space

Comment (iopapamanoglou) [reply src/faebryk/exporters/visualize/interactive_graph.py:68]: but all_gifs = G.nodes

File: src/faebryk/exporters/visualize/interactive_graph.py [hunk 2]
```diff
@@ -0,0 +1,214 @@
...
+    )
+
+    def _not_none(x):
+        return x is not None
+
+    elements = [
+        *(filter(_not_none, (_node(gif) for gif in all_gifs))),
```
Comment (iopapamanoglou) [inline src/faebryk/exporters/visualize/interactive_graph.py:75]: for gif in G

File: src/faebryk/exporters/visualize/interactive_graph.py (hunk 1 above)
Comment (mawildoer) [reply src/faebryk/exporters/visualize/interactive_graph.py]: Ah nice! Thanks mate! Much cleaner!

**Pull request merged**